### PR TITLE
sync email subscriber and profiles keywords field

### PR DESCRIPTION
**Pull request by owocki**
##### Description

sync email subscriber and profiles keywords field per https://github.com/gitcoinco/web/issues/4424

##### Refers/Fixes

https://github.com/gitcoinco/web/issues/4424

##### Testing

tested on my local

### CODE REVIEW

**Comment by owocki** [comment]
its not clear to me that a cronjob is the right place for this... maybe the profile <> emailsubscriber sync could happpen in a presave hook.

**Comment by SaptakS** [comment]
I was thinking more of a post_save hook in both the models.

File: app/dashboard/management/commands/sync_es_profile.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+                counter += 1
+                profile.keywords = keywords
+                profile.save()
+
+        # pull keywords by emailsubscriber into profile
+        for profile in Profile.objects.filter(keywords=[]):
+            es = EmailSubscriber.objects.filter(email=profile.email).exclude(keywords=[]).first()
+            if es:
+                print("2", es.keywords, profile.handle)
+                profile.keywords = es.keywords
```
Comment (danlipert) [inline app/dashboard/management/commands/sync_es_profile.py]: Don't we want to append these keywords rather than overwrite them?

Comment (owocki) [reply app/dashboard/management/commands/sync_es_profile.py]: fe9bdb63292ea26247c79b18ba2f42b17c6e0226

**Comment by owocki** [comment]
just addressed dan's feedback

File: app/dashboard/management/commands/sync_es_profile.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+                if history.data.get('keywords', None):
+                    keywords = history.data['keywords'][0].split(',')
+            if keywords:
+                print("1", counter, profile.handle, keywords)
+                counter += 1
+                profile.keywords = keywords
+                profile.save()
+
+        # pull keywords by emailsubscriber into profile
+        for profile in Profile.objects.filter(keywords=[]):
```
Comment (danlipert) [inline app/dashboard/management/commands/sync_es_profile.py]: Here we are iterating over the profiles that don't have any keywords, but the for loop above assigns keywords to each profile that lacks them if there is a search history object for that user. In this loop, we are combining the email subscriber keywords with the profile keywords, but we are only iterating over profiles that have no keywords.

I think we need to combine the two `for profile in Profile.objects.filter(keywords=[]):` loops so we can get both search history and email subscriber keywords applied. In the current code, adding the search history keywords to a profile essentially skips adding email subscriber keywords to that same profile.

Comment (owocki) [reply app/dashboard/management/commands/sync_es_profile.py]: 32a1a057affae79ff37beebcf5a6914c9d82f085

**Pull request merged**